dask/bag/core.py

```diff
@@ -244,20 +280,39 @@ def __init__(self, dsk, name, npartitions):
         self.npartitions = npartitions
         self.str = StringAccessor(self)
 
-    def map(self, func):
+    def map(self, func, **kwargs):
         """ Map a function across all elements in collection
 
         >>> import dask.bag as db
         >>> b = db.from_sequence(range(5))
         >>> list(b.map(lambda x: x * 10))  # doctest: +SKIP
         [0, 10, 20, 30, 40]
+
+        Any **kwargs are passed through to func along with the (maybe unpacked)
+        item argument(s).  Values in **kwargs must either be a non-dask python
+        literal or a dask.bag.Item.
+
+        If a dask.Bag.Item is used its computed value is passed to the map
+        function.  This means that whatever computations are implied by any
+        Item values block any of the map work.  Example:
+
+        >>> b = db.from_sequence(range(5))
+        >>> b.map(
+        ...     lambda x, total=0: float(x) / total,
+        ...     total=b.sum()).sum().compute()
+        1.0
```

Comment:
This example takes me a moment to grok and so, while it's a fine test, it's a bit odd for documentation.  I wonder if an example like the following might be easier on the reader.

```python
>>> def add(x=0, y=0):
...     return x + y

>>> b = db.from_sequence(range(5))
>>> b.map(add, y=10).compute()
(10, 11, 12, 13, 14)
```